rust/datafusion/src/physical_plan/expressions/coercion.rs

```diff
@@ -44,6 +46,68 @@ pub fn is_numeric(dt: &DataType) -> bool {
         }
 }
 
+/// Get next byte size of the integer number
+fn next_size(size: usize) -> usize {
+    if size < 8_usize {
+        return size * 2;
+    }
+    size
+}
+
+/// Determine if a DataType is float or not
+pub fn is_floating(dt: &DataType) -> bool {
```

Comment:
I think it might be worth putting these functions on  `DataType` itself -- `DataType::is_floating(&self)`, and `DataType::numeric_byte_size(&self)` but we can always do that in a subsequent PR -- it is not needed for this one